homeassistant/components/climate/maxcube.py

```diff
@@ -124,6 +124,21 @@ def set_temperature(self, **kwargs):
                 _LOGGER.error("Setting target temperature failed")
                 return False
 
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+
+        # select device and cube
+        device = self._cubehandle.cube.device_by_rf(self._rf_address)
+        cube = self._cubehandle.cube
+
+        # Set device temperature
+        cube.set_target_temperature(device, temp)
```

Comment:
The cube uses a regular socket to communicate. That is incompatible with asyncio. Regular I/O is not allowed in the asyncio event loop.

This platform has implemented the home assistant sync api, and there's no reason to change that until the cube library supports asyncio.